python-backend/app/api/permits/permit_amendment/resources/permit_amendment.py

```diff
@@ -116,39 +86,78 @@ def post(self, permit_guid=None, permit_amendment_guid=None):
-                description=description)
-
-            for newFile in uploadedFiles:
-                new_pa_doc = PermitAmendmentDocument(
-                    document_name=newFile['fileName'],
-                    document_manager_guid=newFile['document_manager_guid'],
-                    mine_guid=permit.mine_guid,
-                )
-                new_pa.documents.append(new_pa_doc)
-            new_pa.save()
-        except Exception as e:
-            return self.create_error_payload(500, 'Error: {}'.format(e)), 500
-        return new_pa.json()
+        new_pa = PermitAmendment.create(
+            permit,
+            data.get('received_date'),
+            data.get('issue_date'),
+            data.get('authorization_end_date'),
+            data.get('permit_amendment_type_code', 'AMD'),
+            description=data.get('description'))
+
+        uploadedFiles = data.get('uploadedFiles', [])
+        for newFile in uploadedFiles:
+            new_pa_doc = PermitAmendmentDocument(
+                document_name=newFile['fileName'],
+                document_manager_guid=newFile['document_manager_guid'],
+                mine_guid=permit.mine_guid,
+            )
+            new_pa.related_documents.append(new_pa_doc)
+        new_pa.save()
+        return new_pa
+
+
+class PermitAmendmentResource(Resource, UserMixin):
+    parser = reqparse.RequestParser(trim=True)
+
+    parser.add_argument(
+        'permittee_party_guid',
+        type=str,
+        help='GUID of the party that is the permittee for this permit.',
+        location='json')
+    parser.add_argument(
+        'received_date',
+        location='json',
+        type=lambda x: datetime.strptime(x, '%Y-%m-%d') if x else None,
+        store_missing=False)
+    parser.add_argument(
+        'issue_date',
+        location='json',
+        type=lambda x: datetime.strptime(x, '%Y-%m-%d') if x else None,
+        store_missing=False)
+    parser.add_argument(
+        'authorization_end_date',
+        location='json',
+        type=lambda x: datetime.strptime(x, '%Y-%m-%d') if x else None,
+        store_missing=False)
+    parser.add_argument(
+        'permit_amendment_type_code', type=str, location='json', store_missing=False)
+    parser.add_argument(
+        'permit_amendment_status_code', type=str, location='json', store_missing=False)
+    parser.add_argument('description', type=str, location='json', store_missing=False)
+    parser.add_argument('uploadedFiles', type=list, location='json', store_missing=False)
+
+    @api.doc(params={'permit_amendment_guid': 'Permit amendment guid.'})
+    @requires_role_mine_view
+    @api.marshal_with(PERMIT_AMENDMENT_MODEL, code=200)
+    def get(self, permit_guid=None, permit_amendment_guid=None):
+        permit_amendment = PermitAmendment.find_by_permit_amendment_guid(permit_amendment_guid)
+        if not permit_amendment:
+            raise NotFound("Permit Amendment not found.")
+        return permit_amendment
```

Comment:
needs to be checked against the permit as well.

Reply:
I can add that fairly easily, but the cost is a DB hit as the amendment does not have permit_guid to validate against itself. Worth it? I lean no, but I agree it would be better to validate.

Reply:
That was part of the URL pattern discussion is that permit guid needs to matter if it's in the URL.

Reply:
I'm not totally sold on the cost/benefit of being strict, but I'll add it. Conceptually agree, but adds complexity and performance costs.

Reply:
Added. Don't like it as a pattern :)